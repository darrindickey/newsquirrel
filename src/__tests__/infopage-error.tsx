import "jest-dom/extend-expect";
import * as React from "react";
import { render } from "react-testing-library";
import { AxiosResponse } from "../../node_modules/axios";
import Error from "../components/info-pages/error";

interface IError {
  response?: AxiosResponse | null;
}
test("Renders infopage-error correctly", () => {
  const props: IError = {
    response: {
      config: {},
      data: {
        message: "Failed to load the page"
      },
      headers: 'head',
      request: '/',
      status: 400,
      statusText: 'Failure',
    }
  };
  
  const { getByTestId } = render(<Error {...props} />);
  const node = getByTestId("rt-error-page");
  expect(node).toBeInTheDocument();
  expect(node).toMatchSnapshot();
  expect(node).toHaveTextContent("Failed to load the page");
});

