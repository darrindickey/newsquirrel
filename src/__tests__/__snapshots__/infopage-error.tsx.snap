// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders infopage-error correctly 1`] = `
<div
  class="sc-bdVaJa iysoez"
  data-testid="rt-error-page"
>
  Failed to load the page
</div>
`;
